AdminServer/appscale/admin/push_worker_manager.py

```diff
@@ -101,6 +107,22 @@ def update_worker(self, queue_config):
     yield gen.with_timeout(timedelta(seconds=60), start_future,
                            IOLoop.current())
 
+    # Wait for the worker to write its new PID.
+    yield gen.sleep(1)
+    try:
+      with open(pid_location) as pidfile:
+        new_pid = int(pidfile.read().strip())
+    except IOError:
+      new_pid = None
+
+    # Occasionally, Monit will get interrupted during a restart. Retry the
+    # restart if the Celery worker PID is the same.
+    if new_pid == old_pid:
+      logger.warning(
+        '{} worker PID did not change. Restarting it.'.format(self.project_id))
+      yield gen.sleep(1)
+      yield self.update_worker(queue_config)
```

Comment:
I think update in this method will badly work in following scenario:

1. method is called and get old PID 12345
2. it sends command to monit to starts/restarts celery worker
5. sleep, check new PID and get the same 12345, sleep
6. do recursive call
7. get old PID 56789 (it could be changed after initial call checked new PID)
8. `-.-.-` doing the same until process ID is changed strictly during method execution.

Do I missing something? It seems that start/restart operations should be done once and then we should wait until PID is changed.

Reply:
That's a good point. I think the sleep after the log line is not needed.

However, we can't just wait until the PID changes because if Monit is interrupted, it will never change.

Reply:
Oh, are you saying we should wait with an exponential backoff for the pid to change with a longer time limit before retrying the function? That's a good idea.

Reply:
I'm afraid I don't fully understand a possible issue with monit interruption. But I like you solution and it resolves my initial concern.